### PR TITLE
add python 3.10 test

### PR DESCRIPTION
**Pull request by hirosassa**
related to https://github.com/pfnet/pysen/pull/15

I added Python 3.10 entry on supported Python classifier and tests workflow.

### CODE REVIEW

File: assets/Dockerfile [hunk 1]
```diff
@@ -5,9 +5,10 @@ COPY --from=python:3.6.13-alpine3.13 /usr/local/ /usr/local/
 COPY --from=python:3.7.10-alpine3.13 /usr/local/ /usr/local/
 COPY --from=python:3.8.8-alpine3.13 /usr/local/ /usr/local/
 COPY --from=python:3.9.2-alpine3.13 /usr/local/ /usr/local/
+COPY --from=python:3.10.4-alpine3.14 /usr/local/ /usr/local/
```
Comment (hirosassa) [inline assets/Dockerfile:8]: [ask] There's no alpine3.13 images of Python 3.10 on Docker Hub.
Which image should I use here?

https://hub.docker.com/_/python?tab=tags

Comment (Hakuyume) [reply assets/Dockerfile:8]: It is ok to use alpine 3.14. Can you update all images to 3.14?

Comment (hirosassa) [reply assets/Dockerfile:8]: OK!

File: tox.ini [hunk 2]
```diff
@@ -1,9 +1,9 @@
 [tox]
-envlist = py{36,37}-dacite{110,120,150}-isort43-black20, py37-dacite150-isort{43,50,51}-black{19,20}, py{38,39}-dacite150-isort51-black20, development, latest
+envlist = py{36,37}-dacite{110,120,150}-isort43-black20, py37-dacite150-isort{43,50,51}-black{19,20}, py{38,39,310}-dacite150-isort51-black20, development, latest
 
 [testenv]
 deps =
-    pytest >=5.4.0,<6.0.0
+    pytest >=6.2.5,<7.0.0
```
Comment (hirosassa) [inline tox.ini:6]: need to upgrade pytest for python 3.10

ref: https://stackoverflow.com/questions/69528842/an-error-while-trying-to-execute-tests-on-python-3-10-with-pytest

File: assets/Dockerfile (hunk 1 above)
Comment (hirosassa) [reply assets/Dockerfile:8]: done in f698192570aa44ab24565d7ab653d1ea57b498ce

File: assets/Dockerfile [hunk 3]
```diff
@@ -1,16 +1,17 @@
...
+FROM alpine:3.14.6
 
-COPY --from=python:3.6.13-alpine3.13 /usr/local/ /usr/local/
-COPY --from=python:3.7.10-alpine3.13 /usr/local/ /usr/local/
-COPY --from=python:3.8.8-alpine3.13 /usr/local/ /usr/local/
-COPY --from=python:3.9.2-alpine3.13 /usr/local/ /usr/local/
+COPY --from=python:3.6.13-alpine3.14 /usr/local/ /usr/local/
+COPY --from=python:3.7.10-alpine3.14 /usr/local/ /usr/local/
+COPY --from=python:3.8.8-alpine3.14 /usr/local/ /usr/local/
+COPY --from=python:3.9.2-alpine3.14 /usr/local/ /usr/local/
```
Comment (bonprosoft) [inline assets/Dockerfile:7]: It seems those tags aren't available. I think it's okay to upgrade the patch version numbers as:
```suggestion
COPY --from=python:3.8.13-alpine3.14 /usr/local/ /usr/local/
COPY --from=python:3.9.12-alpine3.14 /usr/local/ /usr/local/
```

**Comment by bonprosoft** [comment]
Than you for the updates!
I've built the image and pushed it to quay.io.

https://github.com/pfnet/pysen/blob/b227e2c92d76f8f092da43b91fd80b068001ae74/.github/workflows/pysen-test.yml#L12-L13

Could you please change the image hash to `a37dbee9ecc51ee5240deb275001a459ec177eb4e7b3666ae00088db102c6b9d`?


```diff
diff --git a/.github/workflows/pysen-test.yml b/.github/workflows/pysen-test.yml
index 87778d5..55bb3a1 100644
--- a/.github/workflows/pysen-test.yml
+++ b/.github/workflows/pysen-test.yml
@@ -10,7 +10,7 @@ jobs:
     runs-on: ubuntu-20.04
     timeout-minutes: 15
     container:
-      image: quay.io/pysen/pysen-test@sha256:e6ecc68fda873bf269de16cbed734a627232b88cbdf7054c0520a31a57439738
+      image: quay.io/pysen/pysen-test@sha256:a37dbee9ecc51ee5240deb275001a459ec177eb4e7b3666ae00088db102c6b9d
     steps:
     - uses: actions/checkout@v2
       with:
```

**Comment by hirosassa** [comment]
@bonprosoft Sorry for keeping you waiting. I updated the image hash.

**Comment by bonprosoft** [comment]
I'm so sorry for my late response.

The CI job for the latest commit https://github.com/pfnet/pysen/pull/18/commits/37edfda1862630d386094332a2d71096ce7935ce fails because of the git issue
- https://github.com/pfnet/pysen/runs/6396920277
- https://github.blog/2022-04-12-git-security-vulnerability-announced/

I fixed the issue in #19, so could you rebase this branch from the latest `main`?
Also, I update the pysen-test workflow to build the docker image on each CI job, which means we don't manage the docker image hash on the workflow file anymore.
Could you please revert the change https://github.com/pfnet/pysen/pull/18/commits/37edfda1862630d386094332a2d71096ce7935ce?

I'm sorry to trouble you.

**Comment by hirosassa** [comment]
@bonprosoft Thanks for the reply! I applied it!

**Review by bonprosoft — APPROVED** [review]
Thank you! LGTM!!

**Comment by bonprosoft** [comment]
@Hakuyume friendly ping

**Review by Hakuyume — APPROVED** [review]
LGTM

**Comment by bonprosoft** [comment]
@hirosassa Thank you for your contribution!!

**Pull request merged**